domain/src/test/java/org/oppia/android/domain/devoptions/ShowAllHintsAndSolutionControllerTest.kt

```diff
@@ -0,0 +1,88 @@
+package org.oppia.android.domain.devoptions
+
+import android.app.Application
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [ShowAllHintsAndSolutionController]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = ShowAllHintsAndSolutionControllerTest.TestApplication::class)
+class ShowAllHintsAndSolutionControllerTest {
+
+  @Inject
+  lateinit var showAllHintsAndSolutionController: ShowAllHintsAndSolutionController
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testGetShowAllHintsAndSolution_returnsFalse() {
```

Comment:
```suggestion
  fun testGetShowAllHintsAndSolution_initalState_returnsFalse() {
```

The condition under test is missing I think.

Reply:
Done.